addons/mail/models/mail_thread.py

```diff
@@ -3436,7 +3436,7 @@ def _notify_thread_by_email(self, message, recipients_data, msg_vals=False,
         if force_send and not self.env['ir.mail_server']._disable_send():
             # unless asked specifically, send emails after the transaction to
             # avoid side effects due to emails being sent while the transaction fails
-            if send_after_commit:
+            if not modules.module.current_test and send_after_commit:
```

Comment:
This is not needed, `send_after_commit` has the logic to call `send` during testing.